odl/solvers/nonsmooth/proximal_operators.py

```diff
@@ -1441,6 +1443,231 @@ def _call(self, x, out):
     return ProximalL1L2
 
 
+def proximal_linf(space):
+    """Proximal operator factory of the linf-norm.
+
+    Function for the proximal operator of the functional ``F`` where ``F``
+    is the linf-norm::
+
+        ``F(x) =  \sup_i |x_i|``
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        Domain of F(x).
+
+    Returns
+    -------
+    prox_factory : callable
+        Factory for the proximal operator to be initialized.
+
+    Notes
+    -----
+    The proximal is computed by the Moreau identity and a projection onto an
+    l1-ball [PB2014].
+
+    See Also
+    --------
+    proj_l1 : projection onto l1-ball
+    """
+
+    class ProximalLInf(Operator):
+
+        """Proximal operator of the linf-norm."""
+
+        def __init__(self, sigma):
+            """Initialize a new instance.
+
+            Parameters
+            ----------
+            sigma : positive float
+                Step size parameter
+            """
+            super(ProximalLInf, self).__init__(
+                domain=space, range=space, linear=False)
+            self.sigma = float(sigma)
+
+        def _call(self, x, out):
+            """Apply the operator to ``x`` and stores the result in ``out``."""
+
+            radius = 1
+            proj_l1(x, radius, out)
+            out *= -1
+            out += x
+
+    return ProximalLInf
+
+
+def proj_l1(x, radius=1, out=None):
+    """Projection onto l1-ball.
+
+    Projection onto::
+
+        ``{ x \in X | ||x||_1 \leq r}``
+
+    with `r` being the radius.
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        Space / domain `X`.
+    radius : positive float, optional
+        Radius `r` of the ball.
+
+    Returns
+    -------
+    prox_factory : callable
+        Factory for the proximal operator to be initialized.
+
+    Notes
+    -----
+    The projection onto an l1-ball can be computed by projection onto a
+    simplex, see [D+2008] for details.
+
+    References
+    ----------
+    [D+2008] Duchi, J., Shalev-Shwartz, S., Singer, Y., and Chandra, T.
+    *Efficient Projections onto the L1 -ball for Learning in High dimensions*.
+    ICML 2008, pp. 272–279. http://doi.org/10.1145/1390156.1390191.
+
+    See Also
+    --------
+    proximal_linf : proximal for linf norm
+    proj_simplex : projection onto simplex
+    """
+
+    if out is None:
+        out = x.space.element()
+
+    u = x.ufuncs.absolute()
+    v = x.ufuncs.sign()
+    proj_simplex(u, radius, out)
+    out *= v
+
+    return out
+
+
+def proj_simplex(x, radius=1, out=None):
+    """Projection onto simplex.
+
+    Projection onto::
+
+        ``{ x \in X | x_i \geq 0, \sum_i x_i = r}``
+
+    with `r` being the radius.
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        Space / domain `X`.
+    radius : positive float, optional
+        Radius `r` of simplex.
```

Comment:
\`X\` and \`r\` are dead links.

Reply:
How do I improve on this?

Reply:
Use double-ticks: \`\`X\`\` or \`\`r\`\`
They are merely treated as inline code by Sphinx (the doc generator) and not interpreted as links to class names, glossary terms or similar.